corehq/apps/custom_data_fields/models.py

```diff
@@ -211,6 +211,8 @@ def _migration_sync_to_sql(self, sql_object):
         for field_name in self._migration_get_fields():
             value = getattr(self, field_name)
             setattr(sql_object, field_name, value)
+        if not sql_object.id:
```

Comment:
Fixes https://sentry.io/organizations/dimagi/issues/1754153187/ - once this PR is merged, I'll run the migration with `--verify` and fix any unclean data.